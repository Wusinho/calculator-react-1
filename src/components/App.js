/* eslint-disable no-unused-vars */
import React, { useState } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/Calculate';
import '../index.css';

const App = () => {
  const [value, setValue] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (buttonName) => {
    const result = calculate(value, buttonName);
    setValue({ ...result });
  };
  const { total, next, operation } = value;
  return (
    <div>
      <Display result={(next && next.toString()) || (total && total.toString()) || '0'} partialOp={operation || ''} partialTotal={total || ''} partialNext={next || ''} />
      <ButtonPanel clickHandler={(e) => handleClick(e)} />
    </div>
  );
};

export default App;
