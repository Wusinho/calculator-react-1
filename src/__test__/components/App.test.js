import React from 'react';
import { create } from 'react-test-renderer';
import '@testing-library/jest-dom';
import App from '../../components/App';

test('renders correctly', () => {
  const tree = create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
