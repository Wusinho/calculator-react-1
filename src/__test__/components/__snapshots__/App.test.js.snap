// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="display"
  >
    <h5
      className="resultDisplayer"
    >
      0
    </h5>
    <h5
      className="partialDisplayer"
    >
       
    </h5>
  </div>
  <div
    className="group"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      /
    </button>
  </div>
  <div
    className="group"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </div>
  <div
    className="group"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="group"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="group2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
